File: Partner-Framework/src/main/java/work/slhaf/partner/api/agent/runtime/interaction/AgentInteractionAdapter.java
package work.slhaf.partner.api.agent.runtime.interaction;

import work.slhaf.partner.api.agent.factory.module.pojo.MetaModule;
import work.slhaf.partner.api.agent.runtime.config.AgentConfigManager;
import work.slhaf.partner.api.agent.runtime.interaction.data.AgentInputData;
import work.slhaf.partner.api.agent.runtime.interaction.data.AgentOutputData;
import work.slhaf.partner.api.agent.runtime.interaction.flow.AgentRunningFlow;
import work.slhaf.partner.api.agent.runtime.interaction.flow.entity.RunningFlowContext;

import java.util.List;
import java.util.Map;

public abstract class AgentInteractionAdapter<I extends AgentInputData, O extends AgentOutputData, C extends RunningFlowContext> {

    protected AgentRunningFlow<C> agentRunningFlow = new AgentRunningFlow<>();
    protected Map<Integer, List<MetaModule>> moduleOrderedMap = AgentConfigManager.INSTANCE.getModuleOrderedMap();

    public C call(C finalInputData) {
        return agentRunningFlow.launch(moduleOrderedMap, finalInputData);
    }

    protected abstract O parseOutputData(C outputContext);

    protected abstract C parseInputData(I inputData);

}

File: Partner-Framework/src/main/java/work/slhaf/partner/api/agent/runtime/interaction/AgentInteractionAdapter.kt
package work.slhaf.partner.api.agent.runtime.interaction

import work.slhaf.partner.api.agent.runtime.interaction.data.AgentInputData
import work.slhaf.partner.api.agent.runtime.interaction.data.AgentOutputData
import work.slhaf.partner.api.agent.runtime.interaction.flow.RunningFlowContext

abstract class AgentInteractionAdapter<
        I : AgentInputData,
        O : AgentOutputData,
        C : RunningFlowContext
        > {

    protected val runtime: AgentRuntime = AgentRuntime // 由 AgentContext 持有实例

    fun submit(inputData: I): O {
        val ctx = parseInputData(inputData)

        val result = runtime.submit(ctx)

        return parseOutputData(result)
    }

    protected abstract fun parseOutputData(outputContext: C): O

    protected abstract fun parseInputData(inputData: I): C
}